src/HasTaskHeaders.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stackkit\LaravelGoogleCloudTasksQueue;
+
+interface HasTaskHeaders
+{
+    /** @return array<string, mixed> */
+    public function taskHeaders(): array;
```

Comment:
Nice, I really like this. I think we can remove the `Queue::connection()->setTaskHeaders(...)` so there is a single unified way to add task headers. If tasks require the same set of headers, a trait can be created:

```php
trait WithDefaultTaskHeaders
{
    public function taskHeaders(): array
    {
        return [];
    }
}
```

Much more clear where the task headers are defined rather than a global method somewhere in a service provider.

Reply:
Yeah I personally also want to have 1 way to add task headers, but was doubting if I should remove the `setTaskHeaders()` on the connection or not.

Will do so now.